File: days/1/part1-solution.rb
################################################################################
#
#  Author:  Joe Cohen
#  Mood:    Jolly
#  Problem: https://adventofcode.com/2022/day/1
#  Status:  Solved
#
################################################################################

group_total = largest_total = 0
file = File.open(__dir__ + '/input.txt', 'r')
# file = File.open(__dir__ + '/test-input-1.txt', 'r')

while !file.eof?
  line = file.readline

  if line.strip.empty?
    group_total = 0
  else
    group_total += line.to_i
  end

  largest_total = group_total > largest_total ? group_total : largest_total
end

puts largest_total

File: days/3/part1-solution.ts
/*******************************************************************************
*
*  Author:  Joe Cohen
*  Mood:    Bright
*  Problem: https://adventofcode.com/2022/day/3
*  Status:  Solved
*
*******************************************************************************/
const nReadlines = require('n-readlines');
const inputData = new nReadlines(`${__dirname}/input.txt`);
// const inputData = new nReadlines(`${__dirname}/test-input.txt`);

let line, total = 0;

while (line = inputData.next())
  total += alphaToNumber(findDuplicateItem(line.toString('ascii')));

console.log(`Total: ${total}`);
const used = process.memoryUsage().heapUsed / 1024 / 1024;
console.log(`Memory used: ${Math.round(used * 100) / 100} MB`);

// Helpers
function alphaToNumber(alpha: string | null): number {
  if (!alpha) return 0;

  return 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'.indexOf(alpha) + 1;
}

function findDuplicateItem(items: string): string | null {
  const leftHalf = items.substring(0, items.length / 2);
  const rightHalf = items.substring(items.length / 2);

  for (const item of leftHalf)
    if (rightHalf.includes(item)) return item;

  return null;
}

export {};
